<?xml version="1.0"?>
<launch>
    <node pkg="master_sync_fkie" type="master_sync" name="master_sync" output="log"/>
    <node pkg="master_discovery_fkie" type="master_discovery" name="master_discovery" output="log"/>

    <rosparam file="$(find eurobot_loc)/config/secondary_robot_localization_params.yaml" command="load" ns="secondary_robot"/>
    <rosparam file="$(find eurobot_stm)/config/stm.yaml" command="load" ns="secondary_robot"/>
    <rosparam file="$(find eurobot_main)/config/secondary_robot.yaml" command="load" ns="secondary_robot"/>
    <rosparam file="$(find eurobot_main)/config/bt_params.yaml" command="load" ns="secondary_robot"/>

    <group ns = "secondary_robot">
        <include file="$(find eurobot_nav)/launch/motion_planner_secondary_robot.launch"/>
        <node name="stm_node" pkg="eurobot_stm" type="STM.py" output="log" respawn="true"/>
        <node pkg="urg_node" name="lidar_node" type="urg_node" args="_ip_address:=192.168.0.10" respawn="true" output="log">
            <param name="frame_id" value="/secondary_robot_laser"/>
        </node>
        <node pkg="eurobot_loc" name="particle_filter_node" type="particle_filter_node.py" respawn="true" output="log"/>
        <node pkg="eurobot_stm" name="manipulator_node" type="manipulator.py" respawn="true" output="log"/>
        <node pkg="eurobot_main" name="behavior_tree" type="secondary_robot_bt.py" respawn="true" output="screen"/>
    </group>
</launch>
